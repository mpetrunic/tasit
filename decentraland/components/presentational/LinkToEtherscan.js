import React from "react";
import PropTypes from "prop-types";
import { Button, Icon } from "native-base";
import { showError, openURL } from "@helpers";

const _openEtherscanOf = async address => {
  const url = `https://etherscan.io/address/${address}`;

  try {
    await openURL(url);
  } catch (err) {
    showError(`Unable to open Etherscan for ${address}`);
  }
};

export default function LinkToEtherscan(props) {
  const { account } = props;

  if (account) {
    const { address } = account;
    const openEtherscan = () => _openEtherscanOf(address);
    return (
      <Button transparent onPress={openEtherscan}>
        <Icon name="eye" />
      </Button>
    );
  }

  return null;
}

LinkToEtherscan.propTypes = {
  account: PropTypes.object,
};
