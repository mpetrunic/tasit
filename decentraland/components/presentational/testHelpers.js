import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
};
