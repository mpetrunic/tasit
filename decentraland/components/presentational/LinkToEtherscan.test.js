import React from "react";
import { shallow } from "enzyme";
import LinkToEtherscan from "./LinkToEtherscan";
import { anAction } from "@helpers/testHelpers";

describe("LinkToEtherscan", () => {
  describe("renders the component", () => {
    it("null action", async () => {
      const action = null;
      expect(shallow(<LinkToEtherscan action={action} />)).toMatchSnapshot();
    });

    it("valid action", async () => {
      const action = anAction;
      expect(shallow(<LinkToEtherscan action={action} />)).toMatchSnapshot();
    });
  });
});
