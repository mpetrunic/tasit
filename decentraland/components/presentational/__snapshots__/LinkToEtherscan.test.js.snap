// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkToEtherscan renders the component null action 1`] = `""`;

exports[`LinkToEtherscan renders the component valid action 1`] = `
<Styled(Button)
  onPress={[Function]}
  transparent={true}
>
  <Styled(Icon)
    name="eye"
  />
</Styled(Button)>
`;
